.github/workflows/tests.yml

```diff
@@ -0,0 +1,185 @@
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+      - run: cargo test
+
+  cross-tests:
+    name: Cross tests
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-latest
+            target: mips-unknown-linux-gnu
+            toolchain: stable
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          target: ${{ matrix.target }}
+          toolchain: ${{ matrix.toolchain }}
+          override: true
+      - name: Cache cargo plugins
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/bin/
+          key: ${{ runner.os }}-cargo-plugins
+      - name: Install cross
+        run: cargo install cross || true
+      - name: Test
+        run: cross test --no-fail-fast --target ${{ matrix.target }}
+
+  build:
+    name: Build-only
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        target: [
+          x86_64-sun-solaris,
+          x86_64-unknown-freebsd,
+          x86_64-fuchsia,
+          x86_64-unknown-netbsd,
+          x86_64-unknown-redox,
+          x86_64-fortanix-unknown-sgx,
+        ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          target: ${{ matrix.target }}
+          toolchain: nightly
+          components: rust-src
+          override: true
+      - name: Build
+        run: cargo build --target ${{ matrix.target }}
+
+  build-rdrand:
+    name: Build-only RDRAND
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          components: rust-src
+          override: true
+      - name: Cache cargo plugins
+        uses: actions/cache@v1
+        with:
+          path: ~/.cargo/bin/
+          key: ${{ runner.os }}-cargo-plugins
+      - name: Install xbuild
+        run: cargo install cargo-xbuild || true
+      - name: UEFI
+        run: cargo xbuild --features=rdrand --target x86_64-unknown-uefi
+      - name: Hermit
+        run: cargo xbuild --features=rdrand --target x86_64-unknown-hermit
+      - name: L4Re
+        run: cargo xbuild --features=rdrand --target x86_64-unknown-l4re-uclibc
+      - name: VxWorks
+        run: cargo xbuild --features=rdrand --target x86_64-wrs-vxworks
+
+  clippy-fmt:
+    name: Clippy + rustfmt
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          components: rustfmt, clippy
+      - name: clippy
+        run: cargo clippy --all -- -D warnings
```

Comment:
Do we need `-D warnings` here if it's already set at the top?

Reply:
Hm, not sure. I will check it.

Reply:
Yes, it was redundant. Removed.